multimodal/src/autogluon/multimodal/matcher.py

```diff
@@ -0,0 +1,1715 @@
+from __future__ import annotations
+
+import copy
+import functools
+import json
+import logging
+import operator
+import os
+import pickle
+import shutil
+import sys
+import warnings
+from datetime import timedelta
+from typing import Callable, Dict, List, Optional, Union
+
+import numpy as np
+import pandas as pd
+import pytorch_lightning as pl
+import torch
+import yaml
+from omegaconf import DictConfig, OmegaConf
+from sklearn.model_selection import train_test_split
+from torch import nn
+
+from autogluon.common.utils.log_utils import set_logger_verbosity
+from autogluon.common.utils.utils import setup_outputdir
+from autogluon.core.utils.utils import default_holdout_frac
+
+from . import version as ag_version
+from .constants import (
+    AUTOMM,
+    AUTOMM_TUTORIAL_MODE,
+    BEST,
+    BEST_K_MODELS_FILE,
+    BINARY,
+    CLASSIFICATION,
+    DATA,
+    ENVIRONMENT,
+    FEATURES,
+    GREEDY_SOUP,
+    LABEL,
+    LAST_CHECKPOINT,
+    MATCHER,
+    MAX,
+    MIN,
+    MODEL,
+    MODEL_CHECKPOINT,
+    MULTICLASS,
+    OPTIMIZATION,
+    PAIR,
+    PROBABILITY,
+    QUERY,
+    RESPONSE,
+    TEXT,
+    TRIPLET,
+    UNIFORM_SOUP,
+    Y_PRED,
+    Y_PRED_PROB,
+    Y_TRUE,
+)
+from .data.datamodule import BaseDataModule
+from .data.infer_types import (
+    infer_column_types,
+    infer_label_column_type_by_problem_type,
+    infer_problem_type_output_shape,
+)
+from .optimization.lit_matcher import MatcherLitModule
+from .optimization.utils import get_matcher_loss_func, get_matcher_miner_func, get_metric
+from .utils import (
+    AutoMMModelCheckpoint,
+    CustomUnpickler,
+    LogFilter,
+    apply_log_filter,
+    assign_feature_column_names,
+    average_checkpoints,
+    compute_inference_batch_size,
+    compute_num_gpus,
+    compute_score,
+    create_fusion_data_processors,
+    create_siamese_model,
+    customize_model_names,
+    data_to_df,
+    extract_from_output,
+    get_config,
+    get_local_pretrained_config_paths,
+    get_minmax_mode,
+    infer_dtypes_by_model_names,
+    infer_metrics,
+    infer_precision,
+    init_df_preprocessor,
+    load_text_tokenizers,
+    process_save_path,
+    save_pretrained_model_configs,
+    save_text_tokenizers,
+    select_model,
+    tensor_to_ndarray,
+    try_to_infer_pos_label,
+)
+
+logger = logging.getLogger(AUTOMM)
+
+
+class MultiModalMatcher:
+    """
+    MultiModalMatcher is a deep learning "model zoo" of model zoos. It can automatically build deep learning models that
```

Comment:
It seems here is a duplicate paragraph from `MultimodalPredictor`. We should change the description to better fit what does matcher do.

Reply:
Thanks. Updated the docstrings.